Jamie draft (#2470)
OpenLiberty/blogs: posts/2022-06-31-Open Liberty-&-Semeru-Runtimes-cloud-native-Performance-That-Matters.adoc

---
layout: post

> title: "Open Liberty & Semeru Runtimes, cloud-native Performance That Matters"

Author: ```suggestion
title: "Open Liberty & Semeru Runtimes, cloud-native performance that matters"
```

# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/jamiecoleman92
author_github: https://github.com/jamiecoleman92

> seo-title: TITLE - OpenLiberty.io

Author: ```suggestion
seo-title: Open Liberty & Semeru Runtimes, cloud-native performance that matters - OpenLiberty.io
```

> seo-description: DESCRIPTION

Author: ```suggestion
seo-description: Open Liberty and the IBM Semeru Runtimes JDK are perfectly matched to get the cloud native experience developers expect out of a modern runtime stack, including the best performance and throughput.
```

blog_description: "Explain the advantages of using OpenJ9 with Open Liberty"
open-graph-image: https://openliberty.io/img/twitter_card.jpg
---

> = Open Liberty & Semeru Runtimes, cloud-native Performance That Matters

Author: ```suggestion
= Open Liberty & Semeru Runtimes, cloud-native performance that matters
```

> AUTHOR_NAME <https://github.com/jamiecoleman92>

Author: ```suggestion
Jamie Coleman <https://github.com/jamiecoleman92>
```

:imagesdir: /
:url-prefix:
:url-about: /

> Open Liberty and the IBM Semeru Runtimes JDK are perfectly matched to get the best performance, throughput and the cloud native experience developers expect out of a modern runtime stack. These two technologies have been developed with the mindset that modern developers in the Java community want to get the most out of their Java applications no matter where they are running. 

Author: ```suggestion
Open Liberty and the IBM Semeru Runtimes JDK are perfectly matched to get the cloud native experience developers expect out of a modern runtime stack, including the best performance and throughput. These two technologies were developed for modern developers in the Java community who want to get the most out of their Java applications, no matter where those applications are running. 
```

> The world has moved to a more cloud-native approach that has changed the way software developers fundamentally design and build applications. Open Liberty and the IBM Semeru runtimes JVM have fully embraced this approach to create some of the best performing technologies in this space. 

Author: ```suggestion
The world has moved to a more cloud-native approach that fundamentally changes the way software developers design and build applications. Open Liberty and the IBM Semeru runtimes JVM have fully embraced this approach to create some of the best performing technologies in this space. 
```

> Open Liberty is a runtime developed for this purpose with developer experience at its heart and the OpenJ9 JVM has been built to have characteristics to suit this approach, boasting start-up times of around 1s with Open Liberty https://openliberty.io/blog/2019/10/30/faster-startup-open-liberty[Faster startup time for Open Liberty]. In this blog I will talk about why these two technologies are a great solution for your cloud-native needs with some performance data to back up my claims!

Author: ```suggestion
Open Liberty is a runtime developed for this purpose, with developer experience at its heart. The OpenJ9 JVM is built to suit the cloud-native approach, boasting https://openliberty.io/blog/2019/10/30/faster-startup-open-liberty[start-up times of around 1s with Open Liberty].  In this post, I'll explain why these two technologies are a great solution for your cloud-native needs and provide some performance data to back up my claims!
```

## About OpenJ9/Semeru Runtimes

> OpenJ9 is the JVM that IBM Semeru Runtimes JDK is based upon and IBM customers have been using that JVM for the past 25 years making it very robust. It was given to the Open-source community (Eclipse Foundation) and has since gone through some major changes to make it a great option for your cloud-native Java applications. This JVM has some great performance characteristics that enable Java developers to get the most out of their applications. It can scale from small to large, has some of the best throughput statistics out of all the JVM’s while also providing functionality such as class cache sharing and remote JIT servers to make your applications perform to the highest standards. For some specific performance data on the OpenJ9 JVM you can check this article with testimonies of people that have used it. https://www.eclipse.org/openj9/performance/[Eclipse OpenJ9 performance]

Author: ```suggestion
OpenJ9 is the JVM that IBM Semeru Runtimes JDK is based upon. It is a robust JVM because IBM customers have been using it the past 25 years. It was given to the Open-source community (Eclipse Foundation) and has since gone through some major changes to make it a great option for your cloud-native Java applications. This JVM has some great performance characteristics that enable Java developers to get the most out of their applications. It can scale from small to large and has some of the best throughput statistics of any JVM.  It also provides functions, such as class cache sharing and remote JIT servers, that make your applications perform to the highest standards. For some specific performance data on the OpenJ9 JVM you can check https://www.eclipse.org/openj9/performance[this article with testimonies of people that have used it]. 
```

OpenJ9 vs Hotspot on Performance
image::img/blogs/OpenJ9vsHotspot.jpg[]
image::images/icons/OpenJ9vsHotspot.jpg[OpenJ9 vs Hotspot on Performance]

## Why these technologies are a great match

> Semeru runtimes was once IBM’s JVM, that was donated to the Open-Source community in one of the biggest contributions regarding lines of code that IBM has ever donated along with Open Liberty. It has been used by many of the world’s biggest enterprises due to its robustness and performance. It has changed characteristics over the years to take advantage of what the cloud offers so that it is one of the best performing JVM choices for the cloud. This JVM coupled with Open Liberty has some great performance benefits that we as developers expect from modern cloud-native technologies. 

Author: ```suggestion
Semeru runtimes, along with Open Liberty, is one of the biggest contributions that IBM has ever donated to the open source community. It has been used by many of the world’s biggest enterprises due to its robustness and performance. It has changed characteristics over the years to take advantage of what the cloud offers and is now one of the best performing JVM choices for the cloud. This JVM, coupled with Open Liberty, has some great performance benefits that we as developers expect from modern cloud-native technologies. 
```

> Due to contributions in the open-source communities of these technologies we in the Open Liberty org have been able to make improvements to both the Open Liberty runtime and the OpenJ9 JVM, that dramatically improve things such as start-up time, throughput, ramp-up time, and usability. Start up time is around 1s using technologies such as OpenJ9’s Class Cache sharing https://openliberty.io/blog/2019/10/30/faster-startup-open-liberty.html[Faster startup time for Open Liberty]. Utilizing the OpenJ9 JIT server technology is great if you are working in constrained environments and want great throughput https://developer.ibm.com/articles/jitserver-optimize-your-java-cloud-native-applications/?mhsrc=ibmsearch_a&mhq=jit%20server[JITServer - Optimize your Java cloud-native applications] . Ramp-up time is also important as when your stack is getting warmed up, that is wasted time that is costing you resources. OpenJ9 has great ramp-up time that is great for short lived stacks in the cloud https://www.eclipse.org/openj9/performance/[Eclipse OpenJ9 performance]. In regards to usability, Open Liberty’s dev-mode function enables you to take advantage of OpenJ9’s performance to allow developers to focus on the code and let the runtime automatically deploy your changes so that you can focus on what you love and be more productive https://openliberty.io/blog/2021/04/30/dev-mode-with-aot.html[Faster startup in Open Liberty dev mode].

Author: ```suggestion
Thanks to contributions in the open-source communities of these technologies, we in the Open Liberty org have  made improvements to both the Open Liberty runtime and the OpenJ9 JVM that dramatically improve  start-up time, throughput, ramp-up time,  usability, and more. Start up time is around 1s using technologies such as OpenJ9’s Class Cache sharing. Using the OpenJ9 JIT server technology is great if you are working in constrained environments and want https://developer.ibm.com/articles/jitserver-optimize-your-java-cloud-native-applications/?mhsrc=ibmsearch_a&mhq=jit%20server[great throughput]. Ramp-up time is also important. The time it takes for your stack to get warmed up is wasted time that is costing you resources. OpenJ9 has https://www.eclipse.org/openj9/performance/[great ramp-up time] that is perfect for short lived stacks in the cloud. In regards to usability, https://openliberty.io/blog/2021/04/30/dev-mode-with-aot.html[Open Liberty’s dev-mode function] enables you to take advantage of OpenJ9’s performance, so you can focus on the code and let the runtime automatically deploy your changes. You can focus on what you love and be more productive.
```

## How get started with these technologies
If you wanted to get started with this Java stack, then the easiest way is by downloading the Open Liberty container as this already has the Semeru runtimes JVM pre-installed with some of the functionality mentioned above pre-enabled. You can also get Semeru Runtimes from the Eclipse Adoptium marketplace as mentioned about https://adoptium.net/marketplace/[Marketplace | Adoptium].

## Summary

> From this blog you can hopefully see some of the benefits of using OpenJ9/Semeru Runtimes with the runtime Open Liberty. If you are still building monoliths or creating microservices then you will find performance benefits using this Open source stack. If you want to find more information on why Open Liberty is a great runtime choice you can check out this article https://developer.ibm.com/articles/why-cloud-native-java-developers-love-liberty/[Why cloud-native Java developers love Liberty] or similarly check out this blog for why OpenJ9/Semeru runtimes is a great JVM for your stack https://developer.ibm.com/blogs/introducing-the-ibm-semeru-runtimes/[IBM Developer]

Author: ```suggestion
From this post, you can hopefully see some of the benefits of using OpenJ9/Semeru Runtimes with the Open Liberty runtime . Whether you are still building monoliths or creating microservices, you will find performance benefits using this Open source stack. 

For more information on why Open Liberty is a great runtime choice, check outhttps://developer.ibm.com/articles/why-cloud-native-java-developers-love-liberty/[Why cloud-native Java developers love Liberty]. Similarly, check out https://developer.ibm.com/blogs/introducing-the-ibm-semeru-runtimes/[this post on why OpenJ9/Semeru runtimes is a great JVM for your stack]. 
```

## Next Steps

> If you want to try out this stack, then you can use the Open Liberty cloud hosted guides on our site https://openliberty.io/guides[Open Liberty Guides] that use the latest version of the Open Liberty runtime all running in the cloud on the OpenJ9/Semeru runtimes JVM.

Author: ```suggestion
If you want to try out this stack, check out the https://openliberty.io/guides[Open Liberty cloud hosted guides] that use the latest version of the Open Liberty runtime, all running in the cloud on the OpenJ9/Semeru runtimes JVM.
```
